### ISSUE TITLE
Buy button doesn't respond

### ISSUE THREAD

**Issue by lukechatton** [srikanthonl/ngx-braintree#27]
My implementation gets as far as showing the purchase button (token was returned), but the button does nothing when clicked. No JS errors either.

Implementation:
```html
<ngx-braintree *ngIf="credits == 150" [clientTokenURL]="'-snip-'"
    [createPurchaseURL]="'-snip-'" (paymentStatus)="onPaymentStatus($event)"
    [buttonText]="'Purchase'">
</ngx-braintree>
```

```
Angular CLI: 1.6.0
Node: 8.9.1
OS: darwin x64
Angular: 5.1.1
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

@angular/cdk: 5.0.1
@angular/cli: 1.6.0
@angular/flex-layout: 2.0.0-beta.10-4905443
@angular/material: 5.0.1
@angular-devkit/build-optimizer: 0.0.35
@angular-devkit/core: 0.0.22
@angular-devkit/schematics: 0.0.41
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.9.0
@schematics/angular: 0.1.10
@schematics/schematics: 0.0.10
typescript: 2.4.2
webpack: 3.10.0
```

**Comment by srikanthonl** [comment]
Was your server side createPurchaseURL method invoked when buy is clicked? I see same method, -snip- is being called for token generation and purchasing. If the server side method is invoked when buy is clicked, did you try console logging the response?

**Comment by lukechatton** [comment]
They are different endpoints.

Here is the server side createPurchase endpoint I was testing with. The client never called this though.
```ts
app.post('/credits/braintree/createpurchase/150', this.validate.user, (req, res) => {
            let nonceFromClient = req.body.payment_method_nonce;
            let credits = 150;
            this.createBraintreeSale("1.00", nonceFromClient, (err, result) => {
                if (result) {
                    res.json({})
                } else {
                    res.status(500).json({});
                }
            })
        })
```

onPaymentStatus() never gets called. This is what I have in my shop.component.ts
```ts
onPaymentStatus(response) {
        console.log(response);
}
```

**Comment by lukechatton** [comment]
Could angular be blocking dropin.min.js from being downloaded? Do I have to configure something to allow that?

**Comment by srikanthonl** [comment]
I don’t think angular is blocking because if that’s the case the drop in UI should not be rendered at all but you have it rendered. I’ll look into this and will try to reproduce in what cases the createpurchaseURL is not called AND doesn’t give any error. Please provide more information, if any. Like your component.ts code if possible.

**Comment by srikanthonl** [comment]
Hi Luke,
As you said that the buy button was displayed, did you get the dropin UI rendered as well? The buy button gets displayed as soon as the clientToken is received from your server and the dropin ui gets rendered within moments although sometimes this will take few seconds (depending upon how quickly braintree server responds).

Or is it that you have the dropin UI rendered but nothing happens when you enter the card details and click buy?

Can you pls confirm which one of the above is the behavior?

**Comment by lukechatton** [comment]
The drop in UI is never displayed. Only the buy button.

**Comment by srikanthonl** [comment]
OK. It appears that there is a network/incorrect braintree configuration issue although not sure at this moment if this is an ngx-braintree error or an error in your network. However, a new version of ngx-braintree is published just now (2.2.1) which now throws any error encountered while creating the dropin UI which helps in identifying the root cause. Please update ngx-braintree to the latest version which will now throw the error in console. As I'm unable to reproduce this issue at my end, it will be helpful if you let me know your findings after you have updated the version and tested it.

**Comment by lukechatton** [comment]
Turned out to be a bug with Angular not parsing a text response from Node.js correctly. Submitted a fix over at #29

**Comment by srikanthonl** [comment]
@lukechatton thanks for the fix. Also updated the readme and npm package reflecting this change.
